### DISCUSSION TITLE
Pausing dotnet

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#65654] [General] [answered]
Is there a way to pause dotnet like a virtual machine?
it would be helpful because now I can use it for pause/resume for my games

**Comment by Clockwork-Muse** [comment]
> it would be helpful because now I can use it for pause/resume for my games 

This is completely the wrong way to go about doing a pause system for games, or anything else..

For one thing, if your entire game is running in C#, then you won't be able to _unpause_ the game, because you wouldn't be able to handle any sort of resume event.  Then, even if you could, this would cause problems for things like in-game music, since playback would also be paused.

**Comment by deeprobin** [comment, marked as answer]
You could emit a `int3` instruction ;)

**Reply by Clockwork-Muse** [reply]
Sure, putting the application in debug mode is _totally_ a reasonable way to implement a pause system.... 
/s

**Comment by huoyaoyuan** [comment]
I'd expect task manager to have a feature to pause any non-critical process. The recent Windows update just added throttling feature.

**Reply by Clockwork-Muse** [reply]
... that's not a good way to implement an in-game pause system, either.

**Reply by deeprobin** [reply]
Are you sure you are in the right place :D? If you want to have an ingame pausing, I think you should contact Unity...
